### PR TITLE
Add file based password authenticator plugin

### PR DESCRIPTION
**Pull request by imjalpreet**
Cherry-pick of https://github.com/prestosql/presto/pull/1912 (https://github.com/prestosql/presto/pull/1912)

Co-authored-by: David Phillips <david@acz.org>
Co-authored-by: Rupam Kundu <rkundu@qubole.com>

Related Issue: #15482

This PR also includes moving LDAP Authenticator to a separate package

```
== RELEASE NOTES ==

General Changes
* Add file based password authenticator plugin

### CODE REVIEW

**Comment by dborkar** [comment]
@ashishtadose can you please take a first review? 
@nezihyigitbasi @highker fyi

File: presto-password-authenticators/src/main/java/com/facebook/presto/password/file/FileAuthenticator.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+public class FileAuthenticator
+        implements PasswordAuthenticator
+{
+    private final Supplier<PasswordStore> passwordStoreSupplier;
+
+    @Inject
+    public FileAuthenticator(FileConfig config)
+    {
+        File file = config.getPasswordFile();
```
Comment (ashishtadose) [inline presto-password-authenticators/src/main/java/com/facebook/presto/password/file/FileAuthenticator.java:37]: Need a check to validate if file exists

**Comment by ashishtadose** [comment]
@imjalpreet can you please also add documentation for this plugin

File: presto-password-authenticators/src/main/java/com/facebook/presto/password/file/FileAuthenticator.java (hunk 1 above)
Comment (imjalpreet) [reply presto-password-authenticators/src/main/java/com/facebook/presto/password/file/FileAuthenticator.java:40]: @ashishtadose I have added the validation

Comment (ashishtadose) [reply presto-password-authenticators/src/main/java/com/facebook/presto/password/file/FileAuthenticator.java:40]: looks good now

**Comment by ashishtadose** [comment]
looks good to me.. can you please review it as well @highker

**Review by aweisberg — APPROVED** [review]
+1 thank you
Unrelated but https://github.com/trinodb/trino/commit/9afb92b3ee4743d598510886447c45fc7a78d032#diff-84dd20cc56635d89c0388d00bf9cf21cd1f50425777fdf86e9e43da69d81a94b looks like a nice refactor

**Comment by aweisberg** [comment]
@arhimondr  @rschlussel  can you merge this once the tests pass? It's a pretty straightforward cherry pick.

**Comment by aweisberg** [comment]
Presto accumulo didn't pass, but I don't think it is related. https://github.com/prestodb/presto/issues/15921 is showing up quite a bit now.

**Comment by rschlussel** [comment]
I think it's the same as https://github.com/prestodb/presto/issues/15542.  I haven't figured out the root cause , but it's only shown up on tests that have joins on them.  In the meantime i disable the emptyJoins() test in #15898

**Pull request merged**